From: Holger Assmann <h.assmann@pengutronix.de>
Date: Mon, 21 Jun 2021 10:41:29 +0200
Subject: [PATCH] linux-lxatac: enable usart3 with flow control

Pinmux has been choosen in accordance to the lxatac configuration,
which had not been represented in stm32mp15-pinctrl.dtsi

Signed-off-by: Holger Assmann <h.assmann@pengutronix.de>
---
 arch/arm/boot/dts/stm32mp157c-lxa-tac.dts | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts b/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
index 7f69d3b2be7e..a951ffa8224c 100644
--- a/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
+++ b/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
@@ -207,6 +207,12 @@ &m_can2 {
 	status = "okay";
 };
 
+&usart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&usart3_pins_d>;
+	status = "okay";
+};
+
 &pinctrl {
 	spi5_pins_a: spi5-0 {
 		pins1 {
@@ -222,4 +228,18 @@ pins2 {
 			bias-disable;
 		};
 	};
+	usart3_pins_d: usart3-1 {
+		pins1 {
+			pinmux = <STM32_PINMUX('B', 10, AF7)>, /* USART3_TX */
+				 <STM32_PINMUX('D', 12, AF7)>; /* USART3_RTS */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <0>;
+		};
+		pins2 {
+			pinmux = <STM32_PINMUX('B', 12, AF8)>, /* USART3_RX */
+				 <STM32_PINMUX('D', 11, AF7)>; /* USART3_CTS_NSS */
+			bias-disable;
+		};
+	};
 };
